src/azure-cli/azure/cli/command_modules/appservice/access_restrictions.py

```diff
@@ -57,10 +57,18 @@ def add_webapp_access_restriction(
         access_rules.append(rule_instance)
 
     elif ip_address:
-        rule_instance = IpSecurityRestriction(
-            name=rule_name, ip_address=ip_address,
-            priority=priority, action=action, tag='Default', description=description)
-        access_rules.append(rule_instance)
+        is_exists = False
+        for rule in access_rules:
+            if rule.ip_address == ip_address:
+                is_exists = True
+                break
+        if not is_exists:
+            rule_instance = IpSecurityRestriction(
+                name=rule_name, ip_address=ip_address,
+                priority=priority, action=action, tag='Default', description=description)
+            access_rules.append(rule_instance)
+        else:
+            raise CLIError('Rule is already exists with the specified criteria')
```

Comment:
A few comments here -> I don't think we need the tmp is_exists variable, in Python we can do a lambda filter to check if value exists in object, if we find a value, then we should just raise CLI error saying "Duplicate Rule <RuleName>. already found".

Infact, is there a reason we are not adding this to the validations.py so for add we do the isDuplicateValidation check first before we even enter the execution method here?

Reply:
Made the code changes using Lambda. Regarding adding validation in validations.py file. Validator is already added to the ip_address parameter to check the format (`c.argument('ip_address', help="IP address or CIDR range", validator=validate_ip_address)`). And it is not allowing allowing me to add second validator to pass. So I have added it in function.